### PR TITLE
#410 - Add generic method for mouse events simulation.

### LINKED ISSUE #410: Generalize tests
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/PublicLab.Editor/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update:** In this one, you need to check if there's anything repetitive inside https://github.com/publiclab/Leaflet.DistortableImage/tree/main/test/src files and append the section of code to the `chai` runner object and replace that part of the code in different files with `chai.<your fn name>` as done below.

https://github.com/publiclab/Leaflet.DistortableImage/blob/4a4496f296a49698fe74dd2a08eaf08804efd5f0/test/SpecHelper.js#L14

https://github.com/publiclab/Leaflet.DistortableImage/blob/4a4496f296a49698fe74dd2a08eaf08804efd5f0/test/src/DistortableCollectionSpec.js#L66
 
- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there.

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

### PR DESCRIPTION
**Pull request by Sri-V**
Fixes #410 

* [✅] PR is descriptively titled 📑 and links the original issue above 🔗
* [✅ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt`
* [ ✅] code is in uniquely-named feature branch and has no merge conflicts 📁
* [✅ ] screenshots/GIFs are attached 📎 in case of UI updates - Not applcable
* [✅ ] @mention the original creator of the issue in a comment below for help or for a review
@rexagod 

Adds generic method which takes in the event and provides the corresponding properties where ever required. 

Thanks!

### CODE REVIEW

File: test/SpecHelper.js [hunk 1]
```diff
@@ -2,6 +2,25 @@ beforeEach(function() {
   /* Make expect function available in all tests. */
   window.expect = chai.expect;
 });
+/*
+* Mouse Events is a dictionary of "Events" and their properties.
+* Other events include 'click', 'dblick', 'mouseup', 'mouseover', 'mouseout', 'mousemove'
+* Properties can take in propeties of initMouseEvents().
```
Comment (sashadev-sky) [inline test/SpecHelper.js]:
```suggestion
* Properties can take in properties of initMouseEvents().
```

File: test/SpecHelper.js [hunk 2]
```diff
@@ -2,6 +2,25 @@ beforeEach(function() {
   /* Make expect function available in all tests. */
   window.expect = chai.expect;
 });
+/*
```
Comment (sashadev-sky) [inline test/SpecHelper.js]: Please make sure the * line up for comment blocks! We are working on standardizing our code

File: test/SpecHelper.js [hunk 3]
```diff
@@ -2,6 +2,25 @@ beforeEach(function() {
...
+    type: "mousedown",
+    isShift: false
+  },
+  Click: {
+    type: "click",
+    isShift: false
+  }
+}
 
 /* Chain global testing utilites below to chai*/
```
Comment (sashadev-sky) [inline test/SpecHelper.js]: Lets leave this comment towards the top (position it had before)

File: test/SpecHelper.js [hunk 4]
```diff
@@ -2,6 +2,25 @@ beforeEach(function() {
   /* Make expect function available in all tests. */
   window.expect = chai.expect;
 });
+/*
+* Mouse Events is a dictionary of "Events" and their properties.
+* Other events include 'click', 'dblick', 'mouseup', 'mouseover', 'mouseout', 'mousemove'
+* Properties can take in propeties of initMouseEvents().
+ */
+ chai.mouseEvents = {
+  ShiftDown: {
```
Comment (sashadev-sky) [inline test/SpecHelper.js]: Maybe lets rename this one to `ShiftMouseDown` for clarity? What do you think?

File: test/src/edit/DistortableCollectionEditSpec.js [hunk 5]
```diff
@@ -123,8 +123,9 @@ describe('L.DistortableCollection.Edit', function () {
             expect(map._toolbars).to.be.empty;
 
             // need both bc simulated `mousedown`s don't fire `click` events afterwards like regular user generated `mousedown`s.
-            chai.simulateShiftMousedown(overlay.getElement());
-            chai.simulateClick(overlay.getElement());
+            chai.simulateEvent(overlay.getElement(), chai.mouseEvents.ShiftDown);
+            chai.simulateEvent(overlay.getElement(), chai.mouseEvents.ShiftDown);
```
Comment (sashadev-sky) [inline test/src/edit/DistortableCollectionEditSpec.js]:
```suggestion

```
I think you accidentally added this line twice!! If so, lets delete the extra one

**Comment by Sri-V** [comment]
Hi @sashadev-sky We have implemented the suggested changes. Thanks for guiding us :)

**Pull request merged**